oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/index/ElasticsearchIndexHelperTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.index.elasticsearch.index;
+
+import org.apache.jackrabbit.oak.plugins.index.elasticsearch.ElasticsearchIndexDefinition;
+import org.apache.jackrabbit.oak.plugins.index.elasticsearch.util.ElasticsearchIndexDefinitionBuilder;
+import org.apache.jackrabbit.oak.plugins.index.search.util.IndexDefinitionBuilder;
+import org.apache.jackrabbit.oak.spi.state.NodeState;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class ElasticsearchIndexHelperTest {
+
+    @Test
+    public void multiRulesWithSamePropertyNames() throws IOException {
+        IndexDefinitionBuilder builder = new ElasticsearchIndexDefinitionBuilder();
+        IndexDefinitionBuilder.IndexRule indexRuleA = builder.indexRule("typeA");
+        indexRuleA.property("foo").type("String");
+        IndexDefinitionBuilder.IndexRule indexRuleB = builder.indexRule("typeB");
+        indexRuleB.property("foo").type("String").analyzed();
+        NodeState nodeState = builder.build();
+
+        ElasticsearchIndexDefinition definition =
+                new ElasticsearchIndexDefinition(nodeState, nodeState, "path", "prefix");
+
+        CreateIndexRequest request = ElasticsearchIndexHelper.createIndexRequest(definition);
+        Assert.assertNotNull(request);
```

Comment:
I have not done this because: (a) it is indirectly tested in `ElasticsearchContentTest` (b) there are no POJOs to inspect the mapping and the code looks quite ugly :-)
I have added it now.